src/neo/Ledger/Blockchain.cs

```diff
@@ -263,22 +266,49 @@ private void OnImport(IEnumerable<Block> blocks, bool verify)
             Sender.Tell(new ImportCompleted());
         }
 
+        private void RemoveUnverifiedBlockToCache(uint index)
+        {
+            if (block_cache_unverified.Remove(index, out var entry))
+            {
+                block_cache_unverified_size -= entry.Size;
+            }
+        }
+
         private void AddUnverifiedBlockToCache(Block block)
         {
+            if (block_cache_unverified_size > MaxUnverifiedBlockSize && block.Index > Height + 10)
+            {
+                // Drop last entry if it's higher
+
+                if (block_cache_unverified.Count > 0)
+                {
+                    var max = block_cache_unverified.Keys.Max();
+                    if (max > Height + 10)
+                    {
+                        RemoveUnverifiedBlockToCache(max);
```

Comment:
We should remove hashes from knownHashes, isn't it @erikzhang ?

Reply:
We need to tell TaskManager the index. And Taskmanager needs to be modified.

Reply:
Please, take a look @ShawnYun